src/java/org/apache/cassandra/index/sai/disk/format/OnDiskFormat.java

```diff
@@ -191,4 +194,27 @@ public PerIndexWriter newPerIndexWriter(StorageAttachedIndex index,
      * @return The {@link ByteOrder} for the file associated with the {@link IndexComponentType}
      */
     public ByteOrder byteOrderFor(IndexComponentType component, IndexContext context);
+
+    /**
+     * Encode the given {@link ByteBuffer} into a {@link ByteComparable} object based on the provided {@link AbstractType}
+     * for storage in the in memory trie index. This is valid for term insertion and for encoding search bounds.
+     * @return The encoded {@link ByteComparable} object
+     */
+    public ByteComparable encodeForInMemoryTrie(ByteBuffer input, AbstractType<?> type);
```

Comment:
Maybe add a comment here that we want to control the in-memory format because it is written as-is to disk? So the way we store data in memory affects the disk format.

Reply:
Or maybe we could get rid of this altogether and use `encodeForOnDiskTrie` ?

Reply:
After @blambov's comment regarding historical reasons for distinct encoding, I was able to merge these three encoding methods into just one `encodeForTrie`. It is much cleaner now.